pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -92,28 +91,38 @@ func (p *InstanceTypeProvider) Get(ctx context.Context, provider *v1alpha1.AWS,
 	}
 	var result []cloudprovider.InstanceType
 	for _, i := range instanceTypes {
-		// TODO: move pricing information from the instance type down into offerings
 		instanceTypeName := aws.StringValue(i.InstanceType)
-		price, err := p.pricingProvider.OnDemandPrice(instanceTypeName)
-		if err != nil {
-			// don't warn as this can occur extremely often
-			price = math.MaxFloat64
-		}
-		instanceType := NewInstanceType(ctx, i, kc, price, p.region, provider, p.createOfferings(i, instanceTypeZones[instanceTypeName]))
+		instanceType := NewInstanceType(ctx, i, kc, p.region, provider, p.createOfferings(ctx, i, instanceTypeZones[instanceTypeName]))
 		result = append(result, instanceType)
 	}
 	return result, nil
 }
 
-func (p *InstanceTypeProvider) createOfferings(instanceType *ec2.InstanceTypeInfo, zones sets.String) []cloudprovider.Offering {
-	offerings := []cloudprovider.Offering{}
+func (p *InstanceTypeProvider) createOfferings(ctx context.Context, instanceType *ec2.InstanceTypeInfo, zones sets.String) []cloudprovider.Offering {
+	var offerings []cloudprovider.Offering
 	for zone := range zones {
 		// while usage classes should be a distinct set, there's no guarantee of that
 		for capacityType := range sets.NewString(aws.StringValueSlice(instanceType.SupportedUsageClasses)...) {
 			// exclude any offerings that have recently seen an insufficient capacity error from EC2
-			if _, isUnavailable := p.unavailableOfferings.Get(UnavailableOfferingsCacheKey(*instanceType.InstanceType, zone, capacityType)); !isUnavailable {
-				offerings = append(offerings, cloudprovider.Offering{Zone: zone, CapacityType: capacityType})
+			_, isUnavailable := p.unavailableOfferings.Get(UnavailableOfferingsCacheKey(*instanceType.InstanceType, zone, capacityType))
+			var price float64
+			var err error
+			switch capacityType {
+			case ec2.UsageClassTypeSpot:
+				price, err = p.pricingProvider.SpotPrice(*instanceType.InstanceType, zone)
+			case ec2.UsageClassTypeOnDemand:
+				price, err = p.pricingProvider.OnDemandPrice(*instanceType.InstanceType)
+			default:
+				logging.FromContext(ctx).Errorf("Received unknown capacity type %s for instance type %s", capacityType, *instanceType.InstanceType)
```

Comment:
We used to have this comment:
`// don't warn as this can occur extremely often` in the error handling. Is this potentially really noisy now?

Reply:
Shouldn't be noisy since we are getting the capacity types from the UsageClasses above from AWS. This default case was just there as an extra added measure in case we happen to get a capacity type that we don't know if somehow AWS were to change their SDK